Deploy more smart pointers for member variables in Source/WebKit

rniwa opened the pull request:
#### 5b3959949338a3eac3345c7e84010cd877a0dd6c
<pre>
Deploy more smart pointers for member variables in Source/WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=279427">https://bugs.webkit.org/show_bug.cgi?id=279427</a>

Reviewed by Chris Dumez.

Deployed more smart pointer for member variables in Source/WebKit as warned by clang static analyzer.

* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
(WebKit::RemoteLayerBackingStore::layer const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::backingStoreCollection const):
(WebKit::RemoteLayerBackingStore::encode const):
(WebKit::RemoteLayerBackingStore::drawingRequiresClearedPixels const):
(WebKit::RemoteLayerBackingStore::layer const):
(WebKit::RemoteLayerBackingStore::needsDisplay const):
(WebKit::RemoteLayerBackingStore::performDelegatedLayerDisplay):
(WebKit::RemoteLayerBackingStore::dirtyRepaintCounterIfNecessary):
(WebKit::RemoteLayerBackingStore::paintContents):
(WebKit::RemoteLayerBackingStore::drawInContext):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.h:
(WebKit::RemoteLayerBackingStoreCollection::layerTreeContext const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.mm:
(WebKit::RemoteLayerBackingStoreCollection::sendMarkBuffersVolatile):
(WebKit::RemoteLayerBackingStoreCollection::layerTreeContext const):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithInProcessRenderingBackingStore.mm:
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::createContextAndPaintContents):
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::prepareBuffers):
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::allocateBuffer):
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::prepareToDisplay):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStore.mm:
(WebKit::RemoteLayerWithRemoteRenderingBackingStore::ensureBackingStore):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::Internals::paymentCoordinatorConnection):
(WebKit::WebPageProxy::Internals::paymentCoordinatorBoundInterfaceIdentifier):
(WebKit::WebPageProxy::Internals::getPaymentCoordinatorEmbeddingUserAgent):
(WebKit::WebPageProxy::Internals::paymentCoordinatorPresentingWindow const):
(WebKit::WebPageProxy::Internals::paymentCoordinatorSourceApplicationBundleIdentifier):
(WebKit::WebPageProxy::Internals::paymentCoordinatorSourceApplicationSecondaryIdentifier):
(WebKit::WebPageProxy::Internals::paymentCoordinatorAddMessageReceiver):
(WebKit::WebPageProxy::Internals::paymentCoordinatorRemoveMessageReceiver):
(WebKit::WebPageProxy::Internals::speakingErrorOccurred):
(WebKit::WebPageProxy::Internals::boundaryEventOccurred):
(WebKit::WebPageProxy::Internals::voicesDidChange):
* Source/WebKit/UIProcess/TextCheckerCompletion.cpp:
(WebKit::TextCheckerCompletion::create):
(WebKit::TextCheckerCompletion::TextCheckerCompletion):
(WebKit::TextCheckerCompletion::protectedPage const):
(WebKit::TextCheckerCompletion::spellDocumentTag):
(WebKit::TextCheckerCompletion::didFinishCheckingText const):
(WebKit::TextCheckerCompletion::didCancelCheckingText const):
* Source/WebKit/UIProcess/TextCheckerCompletion.h:
* Source/WebKit/UIProcess/ViewGestureController.cpp:
(WebKit::ViewGestureController::~ViewGestureController):
(WebKit::ViewGestureController::protectedWebPageProxy const):
(WebKit::ViewGestureController::disconnectFromProcess):
(WebKit::ViewGestureController::connectToProcess):
(WebKit::ViewGestureController::willBeginGesture):
(WebKit::ViewGestureController::didEndGesture):
(WebKit::ViewGestureController::canSwipeInDirection const):
(WebKit::ViewGestureController::checkForActiveLoads):
(WebKit::ViewGestureController::PendingSwipeTracker::checkedViewGestureController const):
(WebKit::ViewGestureController::PendingSwipeTracker::protectedWebPageProxy const):
(WebKit::ViewGestureController::PendingSwipeTracker::scrollEventCanBecomeSwipe):
(WebKit::ViewGestureController::PendingSwipeTracker::handleEvent):
(WebKit::ViewGestureController::PendingSwipeTracker::tryToStartSwipe):
(WebKit::ViewGestureController::startSwipeGesture):
(WebKit::ViewGestureController::isPhysicallySwipingLeft const):
(WebKit::ViewGestureController::shouldUseSnapshotForSize):
(WebKit::ViewGestureController::forceRepaintIfNeeded):
(WebKit::ViewGestureController::willEndSwipeGesture):
(WebKit::ViewGestureController::endSwipeGesture):
(WebKit::ViewGestureController::requestRenderTreeSizeNotificationIfNeeded):
(WebKit::ViewGestureController::didCollectGeometryForMagnificationGesture):
(WebKit::ViewGestureController::prepareMagnificationGesture):
(WebKit::ViewGestureController::applyMagnification):
(WebKit::ViewGestureController::endMagnificationGesture):
(WebKit::ViewGestureController::magnification const):
* Source/WebKit/UIProcess/ViewGestureController.h:
(WebKit::ViewGestureController::wheelEventWasNotHandledByWebCore): Deleted.
(WebKit::ViewGestureController::shouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::setShouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::hasActiveMagnificationGesture const): Deleted.
(WebKit::ViewGestureController::setCustomSwipeViews): Deleted.
(WebKit::ViewGestureController::setCustomSwipeViewsTopContentInset): Deleted.
(WebKit::ViewGestureController::setDidMoveSwipeSnapshotCallback): Deleted.
(WebKit::ViewGestureController::backgroundColorForCurrentSnapshot const): Deleted.
(WebKit::ViewGestureController::didFinishNavigation): Deleted.
(WebKit::ViewGestureController::didFailNavigation): Deleted.
(WebKit::ViewGestureController::setSwipeGestureEnabled): Deleted.
(WebKit::ViewGestureController::isSwipeGestureEnabled): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::pause): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::isPaused const): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::hasRemovalCallback const): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::renderTreeSizeThreshold const): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::setRenderTreeSizeThreshold): Deleted.
(WebKit::ViewGestureController::PendingSwipeTracker::shouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::PendingSwipeTracker::setShouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::SwipeProgressTracker::progress const): Deleted.
(WebKit::ViewGestureController::SwipeProgressTracker::direction const): Deleted.
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm:
(WebKit::WebAuthenticatorCoordinatorProxy::performRequest):
* Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.cpp:
(WebKit::WebAuthenticatorCoordinatorProxy::WebAuthenticatorCoordinatorProxy):
(WebKit::WebAuthenticatorCoordinatorProxy::~WebAuthenticatorCoordinatorProxy):
(WebKit::WebAuthenticatorCoordinatorProxy::protectedWebPageProxy const):
(WebKit::WebAuthenticatorCoordinatorProxy::sharedPreferencesForWebProcess const):
(WebKit::WebAuthenticatorCoordinatorProxy::makeCredential):
(WebKit::WebAuthenticatorCoordinatorProxy::getAssertion):
(WebKit::WebAuthenticatorCoordinatorProxy::handleRequest):
* Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.h:
* Source/WebKit/UIProcess/WebBackForwardCache.cpp:
(WebKit::WebBackForwardCache::WebBackForwardCache):
(WebKit::WebBackForwardCache::setCapacity):
* Source/WebKit/UIProcess/WebBackForwardCache.h:
* Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp:
(WebKit::WebFullScreenManagerProxy::WebFullScreenManagerProxy):
(WebKit::WebFullScreenManagerProxy::~WebFullScreenManagerProxy):
(WebKit::WebFullScreenManagerProxy::protectedPage const):
(WebKit::WebFullScreenManagerProxy::sharedPreferencesForWebProcess const):
(WebKit::WebFullScreenManagerProxy::willEnterFullScreen):
(WebKit::WebFullScreenManagerProxy::didEnterFullScreen):
(WebKit::WebFullScreenManagerProxy::willExitFullScreen):
(WebKit::WebFullScreenManagerProxy::didExitFullScreen):
(WebKit::WebFullScreenManagerProxy::setAnimatingFullScreen):
(WebKit::WebFullScreenManagerProxy::requestRestoreFullScreen):
(WebKit::WebFullScreenManagerProxy::requestExitFullScreen):
(WebKit::WebFullScreenManagerProxy::saveScrollPosition):
(WebKit::WebFullScreenManagerProxy::restoreScrollPosition):
(WebKit::WebFullScreenManagerProxy::setFullscreenInsets):
(WebKit::WebFullScreenManagerProxy::setFullscreenAutoHideDuration):
(WebKit::WebFullScreenManagerProxy::beganEnterFullScreen):
* Source/WebKit/UIProcess/WebFullScreenManagerProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::Internals::protectedPage const):
(WebKit::WebPageProxy::Internals::didChooseColor):
(WebKit::WebPageProxy::Internals::didEndColorPicker):
(WebKit::WebPageProxy::Internals::valueChangedForPopupMenu):
(WebKit::WebPageProxy::Internals::setTextFromItemForPopupMenu):
(WebKit::WebPageProxy::Internals::failedToShowPopupMenu):
(WebKit::WebPageProxy::requestCheckingOfString):
(WebKit::WebPageProxy::Internals::setPlaybackTarget):
(WebKit::WebPageProxy::Internals::externalOutputDeviceAvailableDidChange):
(WebKit::WebPageProxy::Internals::setShouldPlayToPlaybackTarget):
(WebKit::WebPageProxy::Internals::playbackTargetPickerWasDismissed):
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::updateBackForwardCacheCapacity):
* Source/WebKit/UIProcess/gstreamer/WebPageProxyGStreamer.cpp:
(WebKit::WebPageProxy::Internals::speakingErrorOccurred):
(WebKit::WebPageProxy::Internals::boundaryEventOccurred):
(WebKit::WebPageProxy::Internals::voicesDidChange):
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp:
(WebKit::ViewGestureController::beginSwipeGesture):
(WebKit::ViewGestureController::handleSwipeGesture):
(WebKit::ViewGestureController::snapshot):
(WebKit::ViewGestureController::draw):
(WebKit::ViewGestureController::removeSwipeSnapshot):
(WebKit::ViewGestureController::setMagnification):
* Source/WebKit/UIProcess/ios/ViewGestureControllerIOS.mm:
(WebKit::ViewGestureController::beginSwipeGesture):
(WebKit::ViewGestureController::willEndSwipeGesture):
(WebKit::ViewGestureController::endSwipeGesture):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::Internals::paymentCoordinatorPresentingViewController):
(WebKit::WebPageProxy::Internals::paymentCoordinatorCTDataConnectionServiceType):
* Source/WebKit/UIProcess/mac/ViewGestureControllerMac.mm:
(WebKit::ViewGestureController::handleMagnificationGestureEvent):
(WebKit::ViewGestureController::handleSmartMagnificationGesture):
(WebKit::ViewGestureController::didCollectGeometryForSmartMagnificationGesture):
(WebKit::ViewGestureController::beginSwipeGesture):
(WebKit::ViewGestureController::handleSwipeGesture):
(WebKit::ViewGestureController::didMoveSwipeSnapshotLayer):
(WebKit::ViewGestureController::resetState):
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::Internals::platformView const):

Canonical link: <a href="https://commits.webkit.org/283489@main">https://commits.webkit.org/283489@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/193d49056b03a7c31d647436a95fd16314d0a89d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66382 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45757 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19003 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70415 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16993 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53556 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17275 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53239 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11853 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69449 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42181 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57465 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33893 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38852 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15868 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60747 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15195 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72118 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10339 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14568 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60564 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10371 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57534 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60878 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8535 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2146 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10065 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41564 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42641 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43824 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42384 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/ViewGestureController.cpp:
```diff
@@ -442,19 +460,19 @@ bool ViewGestureController::PendingSwipeTracker::scrollEventCanBecomeSwipe(Platf
 
-    bool isPinnedToLeft = m_shouldIgnorePinnedState || m_webPageProxy.pinnedState().left();
-    bool isPinnedToRight = m_shouldIgnorePinnedState || m_webPageProxy.pinnedState().right();
+    bool isPinnedToLeft = m_shouldIgnorePinnedState || protectedWebPageProxy()->pinnedState().left();
```
`protectedWebPageProxy()` gets called repeatedly in this function, we should use a local variable instead.

rniwa replied:
Fixed.

cdumez commented on Source/WebKit/UIProcess/ViewGestureController.cpp:
```diff
@@ -666,10 +688,11 @@ void ViewGestureController::requestRenderTreeSizeNotificationIfNeeded()
+        webPageProxy->provisionalPageProxy()->send(Messages::ViewGestureGeometryCollector::SetRenderTreeSizeNotificationThreshold(threshold));
     else
-        m_webPageProxy.legacyMainFrameProcess().send(Messages::ViewGestureGeometryCollector::SetRenderTreeSizeNotificationThreshold(threshold), m_webPageProxy.webPageIDInMainFrameProcess());
+        webPageProxy->legacyMainFrameProcess().send(Messages::ViewGestureGeometryCollector::SetRenderTreeSizeNotificationThreshold(threshold), webPageProxy->webPageIDInMainFrameProcess());
```
`protectedLegacyMainFrameProcess()`

rniwa replied:
Fixed.

cdumez commented on Source/WebKit/UIProcess/ViewGestureController.cpp:
```diff
@@ -690,14 +713,15 @@ void ViewGestureController::didCollectGeometryForMagnificationGesture(FloatRect
-    m_webPageProxy.legacyMainFrameProcess().send(Messages::ViewGestureGeometryCollector::CollectGeometryForMagnificationGesture(), m_webPageProxy.webPageIDInMainFrameProcess());
+    Ref webPageProxy = m_webPageProxy.get();
+    m_magnification = webPageProxy->pageScaleFactor();
+    webPageProxy->legacyMainFrameProcess().send(Messages::ViewGestureGeometryCollector::CollectGeometryForMagnificationGesture(), webPageProxy->webPageIDInMainFrameProcess());
```
`protectedLegacyMainFrameProcess()`

rniwa replied:
Fixed.

cdumez commented on Source/WebKit/UIProcess/ViewGestureController.h:
```diff
@@ -330,6 +333,9 @@ class ViewGestureController : public IPC::MessageReceiver {
         void setShouldIgnorePinnedState(bool ignore) { m_shouldIgnorePinnedState = ignore; }
 
     private:
+        CheckedRef<ViewGestureController> protectedViewGestureController() const;
```
I thought we usually used `checkedFoo()`, not `protectedFoo()` for things that return a CheckedRef / CheckedPtr. I am not sure which way is better but we should be consistent.

rniwa replied:
Oh really? Renamed.

cdumez commented on Source/WebKit/UIProcess/WebBackForwardCache.cpp:
```diff
@@ -51,23 +51,29 @@ WebBackForwardCache::~WebBackForwardCache()
 }
 
-void WebBackForwardCache::setCapacity(unsigned capacity)
+void WebBackForwardCache::setCapacity(WebProcessPool& pool, unsigned capacity)
```
That's a bit of a weird one, I think we should just use `m_processPool` and protect it. It doesn't make much sense to pass a parameter in that we already have. This is a bit of an error prone API. Seems we're trying to optimize an extra Ref out but it doesn't seem like hot code.

rniwa replied:
But that doesn't work because this function gets called during inside WebProcessPool's constructor:
```
Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   CoreFoundation                	       0x187aff4e8 CFRetain.cold.1 + 16
1   CoreFoundation                	       0x187918958 CFRetain + 140
2   WebKit                        	       0x301510a94 API::Object::ref() const + 32 (APIObject.mm:131)
3   WebKit                        	       0x3023dcd4c WTF::DefaultRefDerefTraits<WebKit::WebProcessPool>::ref(WebKit::WebProcessPool&) + 28 (Ref.h:55)
4   WebKit                        	       0x3023dcd18 WTF::Ref<WebKit::WebProcessPool, WTF::RawPtrTraits<WebKit::WebProcessPool>, WTF::DefaultRefDerefTraits<WebKit::WebProcessPool>>::Ref(WebKit::WebProcessPool&) + 40 (Ref.h:86)
5   WebKit                        	       0x3023af788 WTF::Ref<WebKit::WebProcessPool, WTF::RawPtrTraits<WebKit::WebProcessPool>, WTF::DefaultRefDerefTraits<WebKit::WebProcessPool>>::Ref(WebKit::WebProcessPool&) + 40 (Ref.h:87)
6   WebKit                        	       0x3023af678 WebKit::WebBackForwardCache::protectedProcessPool() const + 52 (WebBackForwardCache.cpp:56)
7   WebKit                        	       0x3023af81c WebKit::WebBackForwardCache::setCapacity(unsigned int) + 132 (WebBackForwardCache.cpp:75)
8   WebKit                        	       0x302720ba8 WebKit::WebProcessPool::updateBackForwardCacheCapacity() + 164 (WebProcessPool.cpp:1506)
9   WebKit                        	       0x30271fba8 WebKit::WebProcessPool::WebProcessPool(API::ProcessPoolConfiguration&) + 2288 (WebProcessPool.cpp:306)
10  WebKit                        	       0x30271f0d4 WebKit::WebProcessPool::WebProcessPool(API::ProcessPoolConfiguration&) + 40 (WebProcessPool.cpp:252)
11  WebKit                        	       0x301946864 void API::Object::constructInWrapper<WebKit::WebProcessPool, API::ProcessPoolConfiguration&>(id<WKObject>, API::ProcessPoolConfiguration&) + 64 (APIObject.h:243)
12  WebKit                        	       0x301946808 -[WKProcessPool _initWithConfiguration:] + 140 (WKProcessPool.mm:102)
```

cdumez replied:
That's unfortunate but I don't think we should make the code look worse to silence a warning. We could silence the warning or call `updateBackForwardCacheCapacity()` from `WebProcessPool::create()` & `WKProcessPool _initWithConfiguration:` instead of the constructor

cdumez replied:
Another alternative is to drop `m_processPool` and pass the processPool as parameter like you did. It seems `m_processPool` is only used by `setCapacity()`.

rniwa replied:
Yeah, maybe dropping m_processPool is the best approach. No need to keep a reference around if we can just pass in at call site.

cdumez commented on Source/WebKit/UIProcess/WebBackForwardCache.h:
```diff
@@ -48,7 +48,7 @@ class WebBackForwardCache final : public CanMakeCheckedPtr<WebBackForwardCache>
     ~WebBackForwardCache();
 
-    void setCapacity(unsigned);
+    void setCapacity(WebProcessPool&, unsigned);
```
I don't think we should make this change.

rniwa replied:
So without this change, `WebProcessPool`'s constructor will crash (via `updateBackForwardCacheCapacity()`) because we try to ref `WebProcessPool` before its obj-c object is initialized.

cdumez commented on Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp:
```diff
@@ -65,38 +65,48 @@ WebFullScreenManagerProxy::WebFullScreenManagerProxy(WebPageProxy& page, WebFull
-    if (m_page.isControlledByAutomation()) {
-        if (WebAutomationSession* automationSession = m_page.configuration().processPool().automationSession())
+    if (webPageProxy->isControlledByAutomation()) {
+        if (WebAutomationSession* automationSession = webPageProxy->configuration().processPool().automationSession())
```
`auto*`

rniwa replied:
Changed.

cdumez commented on Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp:
```diff
@@ -126,31 +137,35 @@ void WebFullScreenManagerProxy::didExitFullScreen()
-    if (m_page.isControlledByAutomation()) {
-        if (WebAutomationSession* automationSession = m_page.configuration().processPool().automationSession())
+    if (webPageProxy->isControlledByAutomation()) {
+        if (WebAutomationSession* automationSession = webPageProxy->protectedConfiguration()->processPool().automationSession())
```
`auto*`

rniwa replied:
Changed.

cdumez commented on Source/WebKit/UIProcess/WebProcessPool.cpp:
```diff
@@ -1503,7 +1503,7 @@ void WebProcessPool::updateBackForwardCacheCapacity()
     calculateMemoryCacheSizes(LegacyGlobalSettings::singleton().cacheModel(), dummy, dummy, dummy, dummyInterval, backForwardCacheCapacity);
 
-    checkedBackForwardCache()->setCapacity(backForwardCacheCapacity);
+    checkedBackForwardCache()->setCapacity(*this, backForwardCacheCapacity);
```
I don't think we should make this change.

cdumez reviewed (approved):
LGTM with some changes.

rniwa committed:
Deploy more smart pointers for member variables in Source/WebKit
https://bugs.webkit.org/show_bug.cgi?id=279427

Reviewed by Chris Dumez.

Deployed more smart pointer for member variables in Source/WebKit as warned by clang static analyzer.

* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
(WebKit::RemoteLayerBackingStore::layer const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::backingStoreCollection const):
(WebKit::RemoteLayerBackingStore::encode const):
(WebKit::RemoteLayerBackingStore::drawingRequiresClearedPixels const):
(WebKit::RemoteLayerBackingStore::layer const):
(WebKit::RemoteLayerBackingStore::needsDisplay const):
(WebKit::RemoteLayerBackingStore::performDelegatedLayerDisplay):
(WebKit::RemoteLayerBackingStore::dirtyRepaintCounterIfNecessary):
(WebKit::RemoteLayerBackingStore::paintContents):
(WebKit::RemoteLayerBackingStore::drawInContext):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.h:
(WebKit::RemoteLayerBackingStoreCollection::layerTreeContext const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.mm:
(WebKit::RemoteLayerBackingStoreCollection::sendMarkBuffersVolatile):
(WebKit::RemoteLayerBackingStoreCollection::layerTreeContext const):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithInProcessRenderingBackingStore.mm:
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::createContextAndPaintContents):
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::prepareBuffers):
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::allocateBuffer):
(WebKit::RemoteLayerWithInProcessRenderingBackingStore::prepareToDisplay):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStore.mm:
(WebKit::RemoteLayerWithRemoteRenderingBackingStore::ensureBackingStore):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::Internals::paymentCoordinatorConnection):
(WebKit::WebPageProxy::Internals::paymentCoordinatorBoundInterfaceIdentifier):
(WebKit::WebPageProxy::Internals::getPaymentCoordinatorEmbeddingUserAgent):
(WebKit::WebPageProxy::Internals::paymentCoordinatorPresentingWindow const):
(WebKit::WebPageProxy::Internals::paymentCoordinatorSourceApplicationBundleIdentifier):
(WebKit::WebPageProxy::Internals::paymentCoordinatorSourceApplicationSecondaryIdentifier):
(WebKit::WebPageProxy::Internals::paymentCoordinatorAddMessageReceiver):
(WebKit::WebPageProxy::Internals::paymentCoordinatorRemoveMessageReceiver):
(WebKit::WebPageProxy::Internals::speakingErrorOccurred):
(WebKit::WebPageProxy::Internals::boundaryEventOccurred):
(WebKit::WebPageProxy::Internals::voicesDidChange):
* Source/WebKit/UIProcess/TextCheckerCompletion.cpp:
(WebKit::TextCheckerCompletion::create):
(WebKit::TextCheckerCompletion::TextCheckerCompletion):
(WebKit::TextCheckerCompletion::protectedPage const):
(WebKit::TextCheckerCompletion::spellDocumentTag):
(WebKit::TextCheckerCompletion::didFinishCheckingText const):
(WebKit::TextCheckerCompletion::didCancelCheckingText const):
* Source/WebKit/UIProcess/TextCheckerCompletion.h:
* Source/WebKit/UIProcess/ViewGestureController.cpp:
(WebKit::ViewGestureController::~ViewGestureController):
(WebKit::ViewGestureController::protectedWebPageProxy const):
(WebKit::ViewGestureController::disconnectFromProcess):
(WebKit::ViewGestureController::connectToProcess):
(WebKit::ViewGestureController::willBeginGesture):
(WebKit::ViewGestureController::didEndGesture):
(WebKit::ViewGestureController::canSwipeInDirection const):
(WebKit::ViewGestureController::checkForActiveLoads):
(WebKit::ViewGestureController::PendingSwipeTracker::checkedViewGestureController const):
(WebKit::ViewGestureController::PendingSwipeTracker::protectedWebPageProxy const):
(WebKit::ViewGestureController::PendingSwipeTracker::scrollEventCanBecomeSwipe):
(WebKit::ViewGestureController::PendingSwipeTracker::handleEvent):
(WebKit::ViewGestureController::PendingSwipeTracker::tryToStartSwipe):
(WebKit::ViewGestureController::startSwipeGesture):
(WebKit::ViewGestureController::isPhysicallySwipingLeft const):
(WebKit::ViewGestureController::shouldUseSnapshotForSize):
(WebKit::ViewGestureController::forceRepaintIfNeeded):
(WebKit::ViewGestureController::willEndSwipeGesture):
(WebKit::ViewGestureController::endSwipeGesture):
(WebKit::ViewGestureController::requestRenderTreeSizeNotificationIfNeeded):
(WebKit::ViewGestureController::didCollectGeometryForMagnificationGesture):
(WebKit::ViewGestureController::prepareMagnificationGesture):
(WebKit::ViewGestureController::applyMagnification):
(WebKit::ViewGestureController::endMagnificationGesture):
(WebKit::ViewGestureController::magnification const):
* Source/WebKit/UIProcess/ViewGestureController.h:
(WebKit::ViewGestureController::wheelEventWasNotHandledByWebCore): Deleted.
(WebKit::ViewGestureController::shouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::setShouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::hasActiveMagnificationGesture const): Deleted.
(WebKit::ViewGestureController::setCustomSwipeViews): Deleted.
(WebKit::ViewGestureController::setCustomSwipeViewsTopContentInset): Deleted.
(WebKit::ViewGestureController::setDidMoveSwipeSnapshotCallback): Deleted.
(WebKit::ViewGestureController::backgroundColorForCurrentSnapshot const): Deleted.
(WebKit::ViewGestureController::didFinishNavigation): Deleted.
(WebKit::ViewGestureController::didFailNavigation): Deleted.
(WebKit::ViewGestureController::setSwipeGestureEnabled): Deleted.
(WebKit::ViewGestureController::isSwipeGestureEnabled): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::pause): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::isPaused const): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::hasRemovalCallback const): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::renderTreeSizeThreshold const): Deleted.
(WebKit::ViewGestureController::SnapshotRemovalTracker::setRenderTreeSizeThreshold): Deleted.
(WebKit::ViewGestureController::PendingSwipeTracker::shouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::PendingSwipeTracker::setShouldIgnorePinnedState): Deleted.
(WebKit::ViewGestureController::SwipeProgressTracker::progress const): Deleted.
(WebKit::ViewGestureController::SwipeProgressTracker::direction const): Deleted.
* Source/WebKit/UIProcess/WebAuthentication/Cocoa/WebAuthenticatorCoordinatorProxy.mm:
(WebKit::WebAuthenticatorCoordinatorProxy::performRequest):
* Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.cpp:
(WebKit::WebAuthenticatorCoordinatorProxy::WebAuthenticatorCoordinatorProxy):
(WebKit::WebAuthenticatorCoordinatorProxy::~WebAuthenticatorCoordinatorProxy):
(WebKit::WebAuthenticatorCoordinatorProxy::protectedWebPageProxy const):
(WebKit::WebAuthenticatorCoordinatorProxy::sharedPreferencesForWebProcess const):
(WebKit::WebAuthenticatorCoordinatorProxy::makeCredential):
(WebKit::WebAuthenticatorCoordinatorProxy::getAssertion):
(WebKit::WebAuthenticatorCoordinatorProxy::handleRequest):
* Source/WebKit/UIProcess/WebAuthentication/WebAuthenticatorCoordinatorProxy.h:
* Source/WebKit/UIProcess/WebBackForwardCache.cpp:
(WebKit::WebBackForwardCache::WebBackForwardCache):
(WebKit::WebBackForwardCache::setCapacity):
* Source/WebKit/UIProcess/WebBackForwardCache.h:
* Source/WebKit/UIProcess/WebFullScreenManagerProxy.cpp:
(WebKit::WebFullScreenManagerProxy::WebFullScreenManagerProxy):
(WebKit::WebFullScreenManagerProxy::~WebFullScreenManagerProxy):
(WebKit::WebFullScreenManagerProxy::protectedPage const):
(WebKit::WebFullScreenManagerProxy::sharedPreferencesForWebProcess const):
(WebKit::WebFullScreenManagerProxy::willEnterFullScreen):
(WebKit::WebFullScreenManagerProxy::didEnterFullScreen):
(WebKit::WebFullScreenManagerProxy::willExitFullScreen):
(WebKit::WebFullScreenManagerProxy::didExitFullScreen):
(WebKit::WebFullScreenManagerProxy::setAnimatingFullScreen):
(WebKit::WebFullScreenManagerProxy::requestRestoreFullScreen):
(WebKit::WebFullScreenManagerProxy::requestExitFullScreen):
(WebKit::WebFullScreenManagerProxy::saveScrollPosition):
(WebKit::WebFullScreenManagerProxy::restoreScrollPosition):
(WebKit::WebFullScreenManagerProxy::setFullscreenInsets):
(WebKit::WebFullScreenManagerProxy::setFullscreenAutoHideDuration):
(WebKit::WebFullScreenManagerProxy::beganEnterFullScreen):
* Source/WebKit/UIProcess/WebFullScreenManagerProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::Internals::protectedPage const):
(WebKit::WebPageProxy::Internals::didChooseColor):
(WebKit::WebPageProxy::Internals::didEndColorPicker):
(WebKit::WebPageProxy::Internals::valueChangedForPopupMenu):
(WebKit::WebPageProxy::Internals::setTextFromItemForPopupMenu):
(WebKit::WebPageProxy::Internals::failedToShowPopupMenu):
(WebKit::WebPageProxy::requestCheckingOfString):
(WebKit::WebPageProxy::Internals::setPlaybackTarget):
(WebKit::WebPageProxy::Internals::externalOutputDeviceAvailableDidChange):
(WebKit::WebPageProxy::Internals::setShouldPlayToPlaybackTarget):
(WebKit::WebPageProxy::Internals::playbackTargetPickerWasDismissed):
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::updateBackForwardCacheCapacity):
* Source/WebKit/UIProcess/gstreamer/WebPageProxyGStreamer.cpp:
(WebKit::WebPageProxy::Internals::speakingErrorOccurred):
(WebKit::WebPageProxy::Internals::boundaryEventOccurred):
(WebKit::WebPageProxy::Internals::voicesDidChange):
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp:
(WebKit::ViewGestureController::beginSwipeGesture):
(WebKit::ViewGestureController::handleSwipeGesture):
(WebKit::ViewGestureController::snapshot):
(WebKit::ViewGestureController::draw):
(WebKit::ViewGestureController::removeSwipeSnapshot):
(WebKit::ViewGestureController::setMagnification):
* Source/WebKit/UIProcess/ios/ViewGestureControllerIOS.mm:
(WebKit::ViewGestureController::beginSwipeGesture):
(WebKit::ViewGestureController::willEndSwipeGesture):
(WebKit::ViewGestureController::endSwipeGesture):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::WebPageProxy::Internals::paymentCoordinatorPresentingViewController):
(WebKit::WebPageProxy::Internals::paymentCoordinatorCTDataConnectionServiceType):
* Source/WebKit/UIProcess/mac/ViewGestureControllerMac.mm:
(WebKit::ViewGestureController::handleMagnificationGestureEvent):
(WebKit::ViewGestureController::handleSmartMagnificationGesture):
(WebKit::ViewGestureController::didCollectGeometryForSmartMagnificationGesture):
(WebKit::ViewGestureController::beginSwipeGesture):
(WebKit::ViewGestureController::handleSwipeGesture):
(WebKit::ViewGestureController::didMoveSwipeSnapshotLayer):
(WebKit::ViewGestureController::resetState):
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::Internals::platformView const):

Canonical link: https://commits.webkit.org/283489@main
